reVX/turbine_flicker/turbine_flicker.py

```diff
@@ -0,0 +1,563 @@
+                                         gridcell_height=cls.GRIDCELL_SIZE,
+                                         gridcell_width=cls.GRIDCELL_SIZE,
+                                         gridcells_per_string=1)
+        shadow_flicker = shadow_flicker.create_heat_maps(range(0, 8760),
+                                                         ("time", ))[0]
+
+        return shadow_flicker
+
+    @staticmethod
+    def _check_shadow_flicker_arr(shadow_flicker):
+        """
+        Check to ensure the shadow_flicker array is odd in shape, i.e. both
+        dimensions are odd allowing for a central pixel for the turbine to
+        sit on. Flip 0-axis to mimic the turbine sitting on each building.
+        All flicker pixels will now indicate locations where a turbine would
+        need to be to cause flicker on said building
+
+        Parameters
+        ----------
+        shadow_flicker : ndarray
+            2D array centered on the turbine with the number of flicker hours
+            per "exclusion" pixel
+
+        Returns
+        -------
+        shadow_flicker : ndarray
+            Updated 2D shadow flicker array with odd dimensions if needed
+        """
+        reduce_slice = ()
+        reduce_arr = False
+        for s in shadow_flicker.shape:
+            if s % 2:
+                reduce_slice += (slice(None), )
+            else:
+                reduce_slice += (slice(0, -1), )
+                reduce_arr = True
+
+        if reduce_arr:
+            shape_in = shadow_flicker.shape
+            shadow_flicker = shadow_flicker[reduce_slice]
+            msg = ('Shadow flicker array with shape {} does not have a '
+                   'central pixel! Shade has been reduced to {}!'
+                   .format(shape_in, shadow_flicker.shape))
+            logger.warning(msg)
+            warn(msg)
+
+        return shadow_flicker[::-1]
+
+    @classmethod
+    def _threshold_flicker(cls, shadow_flicker, flicker_threshold=30):
+        """
+        Determine locations of shadow flicker that exceed the given threshold,
+        convert to row and column shifts. These are the locations turbines
+        would need to in relation to building to cause flicker exceeding the
+        threshold value.
+
+        Parameters
+        ----------
+        shadow_flicker : ndarray
+            2D array centered on the turbine with the number of flicker hours
+            per "exclusion" pixel
+        flicker_threshold : int, optional
+            Maximum number of allowable flicker hours, by default 30
+
+        Returns
+        -------
+        row_shifts : ndarray
+            Shifts along axis 0 from building location to pixels to be excluded
+        col_shifts : ndarray
+            Shifts along axis 1 from building location to pixels to be excluded
+        """
+        # ensure shadow_flicker array is regularly shaped
+        shadow_flicker = cls._check_shadow_flicker_arr(shadow_flicker)
```

Comment:
Maybe update the variable + method name because isnt the output now an inverted mapping of exclusion shifts? This is kind of black magic and hard to think about without seeing but if you've checked the output visually then i trust ya and its a great elegant solution!

Reply:
I'll add the plots to the tests, but yeah for matrix math!